core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/resource/BaseMetadataResourceConverter.java

```diff
@@ -220,6 +231,115 @@ protected final void expandDependencyUpgrades(List<R> results) {
 		}
 	}
 	
+	private void expandExtensionOfBranchState(List<R> results) {
+		if (!expand().containsKey(CodeSystem.Expand.EXTENSION_OF_BRANCH_INFO)) {
+			return;
+		}
+		
+		// extensionOf branches are the parent branches of the CodeSystem, so simple branch state calculation is enough
+		for (R result : results) {
+			if (!(result instanceof CodeSystem)) {
+				continue;
+			}
+			
+			CodeSystem codeSystem = (CodeSystem) result;
+			Branch branch = RepositoryRequests.branching().prepareGet(codeSystem.getBranchPath()).build(codeSystem.getToolingId()).execute(context());
+			BranchState branchState = branch.state();
+			codeSystem.setExtensionOfBranchInfo(new BranchInfo(branch.path(), branchState, branch.baseTimestamp(), branch.headTimestamp()));				
+		}
+	}
+
+	private void expandUpgradeOfInfo(List<R> results) {
+		if (!expand().containsKey(CodeSystem.Expand.UPGRADE_INFO)) {
+			return;
+		}
+		
+		final List<ResourceURI> upgradeOfURIs = results.stream()
+				.filter(resource -> resource instanceof CodeSystem)
+				.map(r -> (CodeSystem) r)
```

Comment:
```suggestion
			.filter(CodeSystem.class::isInstance)
			.map(CodeSystem.class::cast)
```
This could be simplfied with the above. Less letters, more readable code!